Gems/Atom/Feature/Common/Code/Source/AuxGeom/AuxGeomDrawQueue.cpp

```diff
@@ -584,6 +585,176 @@ namespace AZ
             AddBox(style, box);
         }
 
+        void AuxGeomDrawQueue::DrawFrustum(
+            const Frustum& frustum,
+            const Color& color,
+            bool drawNormals,
+            DrawStyle style,
+            DepthTest depthTest,
+            DepthWrite depthWrite,
+            FaceCullMode faceCull,
+            int32_t viewProjOverrideIndex)
+        {
+
+            Frustum::CornerVertexArray corners;
+            bool validFrustum = frustum.GetCorners(corners);
+
+            if (validFrustum)
+            {
+                // This helps cut down on clutter below. Replace with a using-enum-declaration when c++20 support is required.
+                enum CornerIndices
+                {
+                    NearTopLeft = Frustum::CornerIndices::NearTopLeft,
+                    NearTopRight = Frustum::CornerIndices::NearTopRight,
+                    NearBottomLeft = Frustum::CornerIndices::NearBottomLeft,
+                    NearBottomRight = Frustum::CornerIndices::NearBottomRight,
+                    FarTopLeft = Frustum::CornerIndices::FarTopLeft,
+                    FarTopRight = Frustum::CornerIndices::FarTopRight,
+                    FarBottomLeft = Frustum::CornerIndices::FarBottomLeft,
+                    FarBottomRight = Frustum::CornerIndices::FarBottomRight,
+                };
+                
+                if (style == DrawStyle::Point)
+                {
+                    RPI::AuxGeomDraw::AuxGeomDynamicDrawArguments drawArgs;
+                    drawArgs.m_verts = corners.data();
+                    drawArgs.m_vertCount = 8;
+                    drawArgs.m_colors = &color;
+                    drawArgs.m_colorCount = 1;
+                    drawArgs.m_depthTest = depthTest;
+                    drawArgs.m_depthWrite = depthWrite;
+                    drawArgs.m_viewProjectionOverrideIndex = viewProjOverrideIndex;
```

Comment:
These settings seem to be common between the 2 paths, perhaps pull them out for clarity.

Reply:
+1

Reply:
I did that because they're different struct types, but since `AuxGeomDynamicIndexedDrawArguments` inherits from `AuxGeomDynamicDrawArguments` I guess there's no harm in just passing `AuxGeomDynamicIndexedDrawArguments` to both.